Plugins/GitUIPluginInterfaces/ApplicationSettings.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace GitUIPluginInterfaces
+{
+    public interface IDetailedSettings
+    {
+        string SmtpServer { get; set; }
+
+        int SmtpPort { get; set; }
+
+        bool SmtpUseSsl { get; set; }
+    }
+
+    public static class DetailedSettingsDefaults
+    {
+        public const string SmtpServerDefault = "smtp.gmail.com";
+        public const int SmtpPortDefault = 465;
+        public const bool SmtpUseSslDefault = true;
```

Comment:
These are not "constants" in the true meaning of the word, and can invariably change.
```suggestion
        public static readonly string SmtpServerDefault = "smtp.gmail.com";
        public static readonly int SmtpPortDefault = 465;
        public static readonly bool SmtpUseSslDefault = true;
```

Reply:
These values will not change.
What are constants for you?

Reply:
`public` constants are compiled into every library. Hence on change, all dependent DLLs would need to be rebuilt and updated together.

Reply:
Class is internal.

`internal sealed class DetailedSettings : IDetailedSettings`

Subsequently, all constants can go to the default configuration and, when used in a cascade, override the value that does not exist.

Reply:
`SmtpPortDefault = 465` this is not a constant. Google can change it any day.

`const` is a compile time artifact, it is burned into the code. So any assembly referencing a constant must be recompiled if `const` value ever changed. That's why `public const` must be avoided at all cost.
`readonly` is a runtime artifact.

Reply:
> SmtpPortDefault = 465 this is not a constant. Google can change it any day.

From this point of view, contant does not exist at all. 😄 

For us, this is the default and is constant. If the port changes, then we change the setting, but the default value will be the same.